src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -248,6 +248,19 @@ internal static void FailFast(string message, Exception? exception, RhFailFastRe
                     outputMessage = message;
                 }
 
+                if (outputMessage != null)
+                {
+                    // Try to save the exception stack trace in a buffer on the stack.  If the exception is too large, we'll truncate it.
+                    const int MaxStack = 1024;
+                    Span<char> exceptionStack = stackalloc char[MaxStack];
+
+                    int length = Math.Max(MaxStack, outputMessage.Length);
+                    outputMessage.AsSpan().Slice(0, length).CopyTo(exceptionStack);
```

Comment:
It may be nice to UTF8 encode the result. It will allow us to keep 2x more information in the typical case.

Reply:
I was worried that encoding to UTF8 in the middle of the exception might be a problem, but if not that makes a lot of sense.